### PR TITLE
Release prep automation and modify Travis config 

### PR DESCRIPTION
**Pull request by pdk27**
*Description* 
* add script to automate release prep - includes 
  1) create and checkout a new branch with the latest tag name
  2) update AEMM and Helm chart versions
  3) commit release prep changes to new branch  
  4) create a PR from the new branch to upstream/master
* modify travis config to run helm e2e tests conditionally
* improve script to create tag for release by sync-ing local tags before tag creation

*Tests*
Sample auto-generated PR - https://github.com/pdk27/amazon-ec2-metadata-mock-1/pull/8
https://github.com/aws/amazon-ec2-metadata-mock/pull/57 (not perfect, but cut to upstream) 
Output of script - shared internally

### CODE REVIEW

File: scripts/create-local-tag-for-release [hunk 1]
```diff
@@ -83,17 +83,33 @@ validate_args() {
     exit 1
 }
 
-# create the new tag
+sync_local_tags_with_remote() {
```
Comment (brycahta) [inline scripts/create-local-tag-for-release:86]: can you explain this a bit? I thought creating local tag is the first step before creating a remote tag

File: scripts/prepare-for-release [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+# This script is NOT intended to be combined with other unmerged local changes
+
+set -o pipefail
+
+REPO_ROOT_PATH="$( cd "$(dirname "$0")"; cd ../; pwd -P )"
+MAKEFILE_PATH=$REPO_ROOT_PATH/Makefile
+LATEST_VERSION=$(make -s -f $MAKEFILE_PATH latest-release-tag | cut -b 2- )
+PREVIOUS_VERSION=$(make -s -f $MAKEFILE_PATH previous-release-tag | cut -b 2- )
+
+# Files to update
```
Comment (brycahta) [inline scripts/prepare-for-release:30]: np: Files *with versions* to update

File: scripts/prepare-for-release [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+## 1) create and checkout a new branch with the latest tag name
+## 2) update AEMM and Helm chart versions
+## 3) commit release prep changes to new branch
+## 4) create a PR from the new branch to upstream/master
+
+## The following files are updated in step (2):
+### - helm charts' Chart.yaml
+### - helm charts' values.yaml
+### - helm charts' README
+### - README.md
```
Comment (brycahta) [inline scripts/prepare-for-release:13]: very helpful comments 👍🏼

File: scripts/prepare-for-release [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+# Prerequisites:
+## 1) git and gh cli (make sure to use gh cli before running this script as authentication is required)
```
Comment (brycahta) [inline scripts/prepare-for-release:16]: soo not driven by GitHub Token?

File: scripts/create-local-tag-for-release (hunk 1 above)
Comment (pdk27) [reply scripts/create-local-tag-for-release:86]: this code vipes out local tags and pulls in all remote tags from upstream. This is done to eliminate the possiblity of local tags being out of sync with remote tags.

remote tag is not created in this script. 

will rename the function to `sync_local_tags_from_remote`

File: scripts/prepare-for-release (hunk 4 above)
Comment (pdk27) [reply scripts/prepare-for-release:16]: thats correct. 
When using gh cli for the first time, when a gh command is run, it directs you to the browser to allow access / authenticate.

**Pull request merged**